docs/proposals/extender-readiness-check.md

```diff
@@ -0,0 +1,78 @@
+# Proposal: Scheduler-extender readiness check
+
+Last updated: 08.07.2021
+
+
+## Abstract
+CSI has no way to verify scheduler-extender deployment status.
+The goal - create an opportunity for user to understand: is scheduler-extender installed completely and able to work or not.
+
+## Background
+Currently, there are 2 patching methods depends on platforms.
+
+- For rke/vanilla kubernetes:
+1. Deploy configuration configmap
+2. Deploy csi-baremetal-patcher daemonset.
+In patcher:
+   - Copy configs from cm to local files on host
+   - Update kubernetes-scheduler manifest
+    
+- For Openshift
+1. Deploy specific scheduler configmap
+2. Update Scheduler recourse
+
+In the both cases patching triggers kube-scheduler restarting process. 
+If kube-scheduler is not restarted, a custom scheduler-extender might not work correctly.
+Restart check is implemented in E2E testing, but it's need to provide this information to user.
+
+## Proposal
+First step - implement CSI CR status field (ready/unready), which is updated in Operator.
+Readiness check - kube-scheduler restarted after patching process.
```

Comment:
On all master nodes? How will it work when you add new master node to the cluster?

Reply:
add in proposal